features/frameworks/greentea-client/source/greentea_test_env.cpp

```diff
@@ -212,53 +209,51 @@ void greentea_notify_coverage_end() {
  *
  *        This function writes the preamble "{{" which is required
  *        for key-value comunication between the target and the host.
- *        This uses a Rawserial object, greentea_serial, which provides
- *        a direct interface to the USBTX and USBRX serial pins and allows
- *        the direct writing of characters using the putc() method.
+ *        This uses greentea_putc which allows the direct writing of characters
+ *        using the write() method.
  *        This suite of functions are provided to allow for serial communication
  *        to the host from within a thread/ISR.
- *
  */
 inline void greentea_write_preamble()
 {
-    greentea_serial->putc('{');
-    greentea_serial->putc('{');
+    greentea_putc('{');
+    greentea_putc('{');
 }
 
 /**
  * \brief Write the postamble characters to the serial port
  *
  *        This function writes the postamble "{{\n" which is required
  *        for key-value comunication between the target and the host.
- *        This uses a Rawserial object, greentea_serial, which provides
- *        a direct interface to the USBTX and USBRX serial pins and allows
- *        the direct writing of characters using the putc() method.
+ *        This uses greentea_putc which allows the direct writing of characters
+ *        using the write() method.
  *        This suite of functions are provided to allow for serial communication
  *        to the host from within a thread/ISR.
  *
  */
 inline void greentea_write_postamble()
 {
-    greentea_serial->putc('}');
-    greentea_serial->putc('}');
-    greentea_serial->putc('\r');
-    greentea_serial->putc('\n');
+    greentea_putc('}');
+    greentea_putc('}');
+    greentea_putc('\r');
+    greentea_putc('\n');
 }
 
 /**
  * \brief Write a string to the serial port
  *
  *        This function writes a '\0' terminated string from the target
  *        to the host. It writes directly to the serial port using the
- *        greentea_serial, Rawserial object.
+ *        greentea_putc which allows the direct writing of characters
+ *        using the write() method.
  *
  * \param str - string value
  *
  */
 inline void greentea_write_string(const char *str)
 {
     while (*str != '\0') {
-        greentea_serial->putc(*str);
+        greentea_putc(*str);
```

Comment:
Why not call `write(STDOUT_FILENO, str, strlen(str))` directly?

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/0ff7ac39b27357207b5124106eabb5d4184672c2..d891e6cac97f8e1ffce4c5e119c8f973fdf15fd3).